ext/standard/array.c

```diff
@@ -6349,9 +6353,24 @@ PHP_FUNCTION(array_key_exists)
 		case IS_NULL:
 			RETVAL_BOOL(zend_hash_exists_ind(ht, ZSTR_EMPTY_ALLOC()));
 			break;
+		case IS_DOUBLE:
+			RETVAL_BOOL(zend_hash_index_exists(ht, zend_dval_to_lval(Z_DVAL_P(key))));
+			break;
+		case IS_FALSE:
+			RETVAL_BOOL(zend_hash_index_exists(ht, 0));
+		case IS_TRUE:
+			RETVAL_BOOL(zend_hash_index_exists(ht, 1));
+			break;
+		case IS_RESOURCE:
+			zend_error(E_WARNING, "Resource ID#%d used as offset, casting to integer (%d)", Z_RES_HANDLE_P(key), Z_RES_HANDLE_P(key));
+			RETVAL_BOOL(zend_hash_index_exists(ht, Z_RES_HANDLE_P(key)));
+			break;
+		case IS_UNDEF:
+			RETVAL_BOOL(zend_symtable_exists_ind(ht, ZSTR_EMPTY_ALLOC()));
```

Comment:
I didn't find a way how to raise the "Undefined variable..." notice here

Reply:
IS_UNDEF can't happen here, you can drop this case.